### ISSUE TITLE
PHP7 compatibility

### ISSUE THREAD

**Issue by josemi-ca** [godaddy/wp-p3-profiler#2]
The plugin doesn't work under PHP 7:

Read here: https://wordpress.org/support/topic/php-7-support-3/

**Comment by danieldunderfelt** [comment]
This would be quite important to get working. Everyone on PHP7 basically SOL because there's no other plugin like P3.

**Comment by frankiejarrett** [comment]
Long story short is this is going to require a complete rewrite of P3.

PHP 7 is pretty amazing, but unfortunately [it broke backwards compatibility](http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.other.func-parameter-modified) with the way `debug_backtrace()` has fundamentally worked in the past, which P3 and other profilers rely on to profile the call stack.

As you can imagine, this is going to take a considerable amount of planning, time and help because we'd essentially be throwing away what we have and starting from scratch. Additionally, whatever approach is decided for the profiler engine to work on PHP 7 would still need to be PHP 5.4 compatible as well, at minimum.

**Comment by JavierSegoviaCordoba** [comment]
Any update about this?

**Comment by xtechgamer735** [comment]
Its a bit out of order that GoDaddy, one of the biggest hosting companies in the business cant spare a few developers for a few days. Or just delete the plugin.

**Comment by nemeth-it** [comment]
Sad, at the current point I could need this plugin a lot.

I'm surprised that there is no warning on the Wordpress Plugin Page (installed and Repository). With such a big incompatibility issue it could become weeks or month until a wordpress-admin would recognizes deprecated or not longer maintained plugins.

So not everything is GoDaddys or the plugin-developers fault. But I would suggest, that they add more hints about this issue on the about-page until it got fixed.

**Comment by MACscr** [comment]
yep. pretty disappointed myself. No other plugin exists that i know of that works like this one too.

**Comment by nemeth-it** [comment]
Working like this one, is true. P3-Profiler was really helpful zu visualize slow plugins really well. But there are still plugins who do also a good job to find the reason why a WP-Installation runs suddenly so slowly.

I use [Query Monitor](https://de.wordpress.org/plugins/query-monitor/) now, as long it is activated, it shows in the too-bar on each page-load what and how many stuff got loaded and slowed down your page. (Querys from Themes, Plugins, Hooks and so on.)

**Comment by webdados** [comment]
RIP P3 :-(

**Comment by mkormendy** [comment]
> Long story short is this is going to require a complete rewrite of P3.
> 
> PHP 7 is pretty amazing, but unfortunately [it broke backwards compatibility](http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.other.func-parameter-modified) with the way `debug_backtrace()` has fundamentally worked in the past, which P3 and other profilers rely on to profile the call stack.
> 
> As you can imagine, this is going to take a considerable amount of planning, time and help because we'd essentially be throwing away what we have and starting from scratch. Additionally, whatever approach is decided for the profiler engine to work on PHP 7 would still need to be PHP 5.4 compatible as well, at minimum.

So stop wasting time and start rewriting. 😀

**Comment by davidfavor** [comment]
https://wordpress.org/about/requirements - state minimum PHP version is now 7.3 so seems like targeting support for 5.4 - 5.6 might require a huge effort, for very little gain.

Just a thought.

**Comment by nozimm** [comment]
1. open the file [plugins/p3-profiler/templates/template.php],and prepend the code below:
`<script>
    jQuery.extend({
	browser: function() 
	{
		var
	    rwebkit = /(webkit)\/([\w.]+)/,
	    ropera = /(opera)(?:.*version)?[ \/]([\w.]+)/,
	    rmsie = /(msie) ([\w.]+)/,
	    rmozilla = /(mozilla)(?:.*? rv:([\w.]+))?/,    
	    browser = {},
	    ua = window.navigator.userAgent,
	    browserMatch = uaMatch(ua);
 
	    if (browserMatch.browser) {
	        browser[browserMatch.browser] = true;
	        browser.version = browserMatch.version;
	    }
	    return { browser: browser };
	},
});
 
function uaMatch(ua) 
{
        ua = ua.toLowerCase();
 
        var match = rwebkit.exec(ua)
                    || ropera.exec(ua)
                    || rmsie.exec(ua)
                    || ua.indexOf("compatible") < 0 && rmozilla.exec(ua)
                    || [];
        return {
            browser : match[1] || "",
            version : match[2] || "0"
        };
}

</script>`
2. open the file [plugins/p3-profiler/templates/callouts.php],and replace ‘.live’ to ‘.on’

Solution by huajun102: https://wordpress.org/support/topic/it-cant-run-let-me-tell-you-how-to-fix-it/

**Comment by webdados** [comment]
@nozimm Not related to this issue, at all.

**Comment by nozimm** [comment]
> @nozimm Not related to this issue, at all.

woops, wrong tab

**Comment by MACscr** [comment]
> Long story short is this is going to require a complete rewrite of P3.
> 
> PHP 7 is pretty amazing, but unfortunately [it broke backwards compatibility](http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.other.func-parameter-modified) with the way `debug_backtrace()` has fundamentally worked in the past, which P3 and other profilers rely on to profile the call stack.
> 
> As you can imagine, this is going to take a considerable amount of planning, time and help because we'd essentially be throwing away what we have and starting from scratch. Additionally, whatever approach is decided for the profiler engine to work on PHP 7 would still need to be PHP 5.4 compatible as well, at minimum.

like anyone that cares about performance is running anything less than php 7

**Comment by gabriel-munteanu** [comment]
I did a fix for this issue. There is a pull request for this: https://github.com/wpmudev/wp-p3-profiler/pull/8. Meanwhile you can download the plugin with the fix included from here: https://github.com/gabriel-munteanu/wp-p3-profiler/blob/master/p3-profiler.zip
